core/src/commitment.rs

```diff
@@ -75,6 +75,37 @@ impl BlockCommitmentCache {
     pub fn get_rooted_block_with_commitment(&self, minimum_stake_percentage: f64) -> Option<u64> {
         self.get_block_with_depth_commitment(MAX_LOCKOUT_HISTORY - 1, minimum_stake_percentage)
     }
+
+    pub fn get_confirmation_count(
+        &self,
+        slot: Slot,
+        minimum_stake_percentage: f64,
+    ) -> Option<usize> {
+        self.get_block_commitment(slot).map(|block_commitment| {
+            if let Some((index, _stake)) =
+                block_commitment
+                    .commitment
+                    .iter()
+                    .enumerate()
+                    .find(|(_, &stake)| {
```

Comment:
Looks like this is searching backwards

Reply:
There will be at most one index with 66+% of the stake (see comment below), so there isn't really a "backwards".